### PR TITLE
Vueify Admin Forms and Quota grids

### PR DESCRIPTION
**Pull request by guerler**
Requires #17126. This PR converts the forms and quota grid to Vue, additionally it contains fixes such as additional confirmation alerts for the user grid and adjustments to the filter query.

image

image

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/controllers/admin.py [hunk 1]
```diff
@@ -649,7 +545,7 @@ def rename_quota(self, trans, payload=None, **kwd):
         quota = get_quota(trans, id)
         if trans.request.method == "GET":
             return {
-                "title": "Change quota name and description for '%s'" % util.sanitize_text(quota.name),
+                "title": "Change quota name and description for '%s'" % quota.name,
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/controllers/admin.py]:
```suggestion
                "title": f"Change quota name and description for '{quota.name}'",
```

File: lib/galaxy/webapps/galaxy/controllers/admin.py [hunk 2]
```diff
@@ -718,7 +614,7 @@ def edit_quota(self, trans, payload=None, **kwd):
         quota = get_quota(trans, id)
         if trans.request.method == "GET":
             return {
-                "title": "Edit quota size for '%s'" % util.sanitize_text(quota.name),
+                "title": "Edit quota size for '%s'" % quota.name,
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/controllers/admin.py]:
```suggestion
                "title": f"Edit quota size for '{quota.name}'",
```

File: lib/galaxy/webapps/galaxy/controllers/admin.py [hunk 3]
```diff
@@ -753,7 +649,7 @@ def set_quota_default(self, trans, payload=None, **kwd):
             for typ in trans.app.model.DefaultQuotaAssociation.types.__members__.values():
                 default_options.append((f"Yes, {typ}", typ))
             return {
-                "title": "Set quota default for '%s'" % util.sanitize_text(quota.name),
+                "title": "Set quota default for '%s'" % quota.name,
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/controllers/admin.py]:
```suggestion
                "title": f"Set quota default for '{quota.name}'",
```

File: lib/galaxy/webapps/galaxy/controllers/admin.py [hunk 4]
```diff
@@ -902,7 +798,7 @@ def rename_role(self, trans, payload=None, **kwd):
         role = get_role(trans, id)
         if trans.request.method == "GET":
             return {
-                "title": "Change role name and description for '%s'" % util.sanitize_text(role.name),
+                "title": "Change role name and description for '%s'" % role.name,
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/controllers/admin.py]:
```suggestion
                "title": f"Change role name and description for '{role.name}'",
```

File: lib/galaxy/webapps/galaxy/controllers/admin.py [hunk 5]
```diff
@@ -1012,7 +908,7 @@ def rename_group(self, trans, payload=None, **kwd):
         group = get_group(trans, id)
         if trans.request.method == "GET":
             return {
-                "title": "Change group name for '%s'" % util.sanitize_text(group.name),
+                "title": "Change group name for '%s'" % group.name,
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/controllers/admin.py]:
```suggestion
                "title": f"Change group name for '{group.name}'",
```

File: lib/galaxy/webapps/galaxy/controllers/forms.py [hunk 6]
```diff
@@ -194,7 +187,7 @@ def edit_form(self, trans, payload=None, **kwd):
                 {"name": "required", "label": "Required", "type": "boolean", "value": False},
             ]
             form_dict = {
-                "title": "Edit form for '%s'" % (util.sanitize_text(latest_form.name)),
+                "title": "Edit form for '%s'" % latest_form.name,
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/controllers/forms.py]:
```suggestion
                "title": f"Edit form for '{latest_form.name}'",
```

File: lib/galaxy/webapps/galaxy/controllers/forms.py [hunk 7]
```diff
@@ -303,28 +296,6 @@ def save_form_definition(self, trans, form_id=None, payload=None, **kwd):
...
-    @web.require_admin
-    def _undelete_form(self, trans, ids):
-        for form_id in ids:
-            form = get_form(trans, form_id)
-            form.deleted = False
-            trans.sa_session.add(form)
-            with transaction(trans.sa_session):
-                trans.sa_session.commit()
-        return ("Undeleted %i form(s)." % len(ids), "done")
-
```
Comment (davelopez) [inline lib/galaxy/webapps/galaxy/controllers/forms.py]: This *Utility methods* block can be removed and the `get_form` utility can be then imported from `galaxy.managers.forms`

**Review by davelopez — APPROVED** [review]
Looks good, just some minor formatting comments.

I don't know how it should be, but it seems you can `undelete` a quota that has been previously purged, should we just completely remove that quota from the database after purging?

File: lib/galaxy/webapps/galaxy/controllers/forms.py (hunk 7 above)
Comment (guerler) [reply lib/galaxy/webapps/galaxy/controllers/forms.py]: Thank you!

**Review by davelopez — APPROVED** [review]

**Pull request merged**